File: javascript/find_shortest_string.js
function findShortestString(arr) {
  let shortest = arr[0];
  arr.forEach((word) => {
    if (word.length < shortest.length) {
      shortest = word;
    }
  });
  return shortest;
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'a'");
  console.log('=>', findShortestString(['aaa', 'a', 'bb', 'ccc']));

  console.log('');

  console.log("Expecting: 'hi'");
  console.log('=>', findShortestString(['cat', 'hi', 'dog', 'an']));

  console.log('');

  console.log("Expecting: 'lily'");
  console.log(
    '=>',
    findShortestString(['flower', 'juniper', 'lily', 'dadelion'])
  );

  // BENCHMARK HERE
}

module.exports = findShortestString;

// Please add your pseudocode to this file
// shortest word = arr[0]
// for word in arr:
// if word shorter than shortest word:
// shortest word = word
// return shortest word

// And a written explanation of your solution

// Iterate over the array and compare each word to the current shortest word.

File: ruby/find_shortest_string.rb
def find_shortest_string(arr)
  shortest = arr[0]
  arr.each do |string|
    if string.length < shortest.length then shortest = string end
  end
  shortest
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'a'"
  puts "=>", find_shortest_string(['aaa', 'a', 'bb', 'ccc'])

  puts

  puts "Expecting: 'hi'"
  puts "=>", find_shortest_string(['cat', 'hi', 'dog', 'an'])

  puts

  puts "Expecting: 'lily'"
  puts "=>", find_shortest_string(['flower', 'juniper', 'lily', 'dadelion'])

  # Don't forget to add your own!

  # BENCHMARK HERE
end

# Please add your pseudocode to this file

# loop through strings
# save if shorter than previous lowest
# return

# And a written explanation of your solution
